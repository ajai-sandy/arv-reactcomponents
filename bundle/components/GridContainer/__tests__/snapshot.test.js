import React from "react";
import { mount, shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import {
  GridContainerSuccessTest,
  GridContainerErrorTest,
} from "../test.component";

describe("Root Component Tests", () => {
  it("Matches Shallow SnapShot", () => {
    expect(
      shallowToJson(shallow(<GridContainerSuccessTest />)),
    ).toMatchSnapshot();
    expect(
      shallowToJson(shallow(<GridContainerErrorTest />)),
    ).toMatchSnapshot();
  });

  it("Matches Mounted SnapShot", done => {
    const successMounted = mount(<GridContainerSuccessTest />);
    const errorMounted = mount(<GridContainerErrorTest />);

    setTimeout(() => {
      expect(successMounted).toMatchSnapshot();

      expect(errorMounted).toMatchSnapshot();

      done();
    }, 300);
  });
});
