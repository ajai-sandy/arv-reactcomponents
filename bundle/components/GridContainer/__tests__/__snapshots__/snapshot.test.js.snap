// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component Tests Matches Mounted SnapShot 1`] = `
<GridContainerSuccessTest>
  <GridContainer
    className=""
    strict={false}
  >
    <div
      className="nwc-grid-container  "
    >
      <GridRow
        className=""
      >
        <div
          className="nwc-grid-row "
        >
          <GridColumn
            className=""
          >
            <div
              className="nwc-grid-col "
            >
              Hello
            </div>
          </GridColumn>
        </div>
      </GridRow>
    </div>
  </GridContainer>
</GridContainerSuccessTest>
`;

exports[`Root Component Tests Matches Mounted SnapShot 2`] = `
<GridContainerErrorTest>
  <GridContainer
    className=""
    strict={false}
  >
    <div
      className="nwc-grid-container  "
    >
      <div>
        Hello
      </div>
    </div>
  </GridContainer>
</GridContainerErrorTest>
`;

exports[`Root Component Tests Matches Shallow SnapShot 1`] = `
<GridContainer
  className=""
  strict={false}
>
  <GridRow
    className=""
  >
    <GridColumn
      className=""
    >
      Hello
    </GridColumn>
  </GridRow>
</GridContainer>
`;

exports[`Root Component Tests Matches Shallow SnapShot 2`] = `
<GridContainer
  className=""
  strict={false}
>
  <div>
    Hello
  </div>
</GridContainer>
`;
